modules/platforms/cpp/client/src/ignite_client.cpp

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <thread>
+
+#include "common/ignite_error.h"
+#include "ignite/ignite_client.h"
+#include "ignite_client_impl.h"
+
+namespace ignite
+{
+
+std::future<IgniteClient> IgniteClient::startAsync(IgniteClientConfiguration configuration,
+    std::chrono::milliseconds timeout)
+{
+    return std::async(std::launch::async, [cfg = std::move(configuration), timeout] () mutable {
```

Comment:
This launches a new thread to execute a synchronous `start` method, which calls async `impl->start()` and then blocks the thread with `wait_for`.

Ideally we should not block the user thread in async methods.

Reply:
Ideally yes )